pkg/registry/extensions/deployment/strategy.go

```diff
@@ -90,6 +92,7 @@ func (deploymentStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, o
 		!reflect.DeepEqual(newDeployment.Annotations, oldDeployment.Annotations) {
 		newDeployment.Generation = oldDeployment.Generation + 1
 	}
+	checkAndSetMaxUnavailableForRollingUpdate(newDeployment)
```

Comment:
We are not supposed to mutate the spec for handling this one. Instead, it should be handled in the controller.